ara/plugins/callback/ara_default.py

```diff
@@ -660,7 +660,7 @@ def _load_result(self, result, status, **kwargs):
             ignore_errors=kwargs.get("ignore_errors", False) or False,
         )
 
-        if task["action"] in ["setup", "gather_facts"] and "ansible_facts" in results:
+        if task["action"].split(".")[-1] in ["setup", "gather_facts"] and "ansible_facts" in results:
```

Comment:
As @flowerysong mentioned, this could be error prone even if under rare circumstances, I think.

I also agree with you that the long static list of items wouldn't be the prettiest thing.
Should we just put the list as a constant at the top of the file ? Like before the ``DOCUMENTATION`` block:

```python
# Task modules for which ara should save host facts
ANSIBLE_SETUP_MODULES = [
    "setup",
    "ansible.builtin.setup",
    "ansible.legacy.setup",
    "gather_facts",
    "ansible.builtin.gather_facts",
    "ansible.legacy.setup",
]

DOCUMENTATION = """
# ...
```

and then the code above would become something like:
```python
if task["action"] in ANSIBLE_SETUP_MODULES and "ansible_facts" in results:
```

Reply:
Do you think we might need that for another module again in the future? Then we could also implement the helper function which @flowerysong mentioned.

Reply:
If I am looking at the right [function](https://github.com/ansible/ansible/blob/5728d72cda94e24314a95befb0130b9e773f1035/lib/ansible/utils/fqcn.py#L21-L33), it prefixes ansible.builtin/ansible.legacy to the module names.

I wouldn't expect ansible to come up with another prefix and I don't know whether there'd be other modules to add in the future.

The approach I suggested wouldn't bother me at all but if we really wanted to, we could also just declare setup/gather_facts and then import ``add_internal_fqcns`` from ansible.utils and use that.

It exists as far back as ansible 2.9, though it would be a breaking change for ansible 2.8 but it's also long EOL.

Reply:
thus I did your suggested approach but defined the `ANSIBLE_SETUP_MODULES` below the `DOCUMENTATION` part. 
Thought that it might good if documentation part is still on the top

Reply:
Sure.